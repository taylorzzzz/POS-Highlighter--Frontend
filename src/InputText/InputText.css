@import '../colors.css';

.InputText {
  position: relative;
  width: 100%;
  min-height: 100%;
  box-sizing: border-box;
  padding: 0;
  margin-left: 10px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  z-index: 1;
}

textarea {
  resize: none;
  padding: 0px;
  width: 100%;
  min-height: 80%;
  font-size: 14px;
  font-weight: 100;
  line-height: 1.5em;
  color: var(--color-text);
  background: var(--color-bg);
  outline: none;
  border: 1px solid #eee;
  border-radius: 3px;
  box-shadow: 0px 2px 10px 4px rgba(0,0,0,.02);
}
textarea::selection {
  color: var(--color-bg);
  background: var(--color-text);
}

.InputText button {
  color: var(--color-bg);
  background: var(--color-text);
  padding: 5px 5px;
  margin-right: 0px;
  margin-top: 10px;
  border: none;
  cursor: pointer;
  transition: all .2s;
  outline: none;
  border-radius: 3px;
  box-shadow: 2px 2px 4px rgba(0,0,0,.25);
}

.InputText button:active {
  box-shadow: 1px 1px 3px rgba(0,0,0,.25);
}