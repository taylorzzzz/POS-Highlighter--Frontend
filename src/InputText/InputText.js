import React from 'react';

import './InputText.css';

function InputText({text, handleTextChange, submitText}) {
  
  return (
    <div className="InputText">

      <textarea 
        onChange={handleTextChange}
        value={text} autoFocus="true">
      </textarea>

      <button 
        onClick={submitText}>
          Highlight
      </button>

    </div>
    
  )
}

export default InputText;