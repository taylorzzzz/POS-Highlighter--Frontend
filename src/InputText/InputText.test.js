import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import InputText from './InputText';

test('InputText renders properly', () => {
    const tree = renderer.create(<InputText />).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('InputText handleTextChange events properly', () => {

    const textChangeSpy = jest.fn();

    const tree = shallow(<InputText handleTextChange={textChangeSpy}/>);

    const textarea = tree.find('textarea');

    expect(textChangeSpy).toHaveBeenCalledTimes(0);

    textarea.simulate('change');
    
    expect(textChangeSpy).toHaveBeenCalledTimes(1);
})

test('InputText submitText events properly', () => {

    const testText = "I am the test text.";

    const submitTextSpy = jest.fn();

    const tree = shallow(<InputText submitText={submitTextSpy} />);

    const button = tree.find('button');

    expect(submitTextSpy).toHaveBeenCalledTimes(0);

    button.simulate('click');
    
    expect(submitTextSpy).toHaveBeenCalledTimes(1);
})

test('InputText text prop passed to textarea properly', () => {

    const testText = "I am the test text.";

    const tree = shallow(<InputText text={testText}/>);

    const textarea = tree.find('textarea');

    expect(textarea.props().value).toBe(testText);
})