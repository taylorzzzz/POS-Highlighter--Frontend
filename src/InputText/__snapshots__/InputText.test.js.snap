// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders properly 1`] = `
<div
  className="InputText"
>
  <textarea
    autoFocus="true"
    onChange={undefined}
    value={undefined}
  />
  <button
    onClick={undefined}
  >
    Highlight
  </button>
</div>
`;
